gmso/formats/lammpsdata.py

```diff
@@ -1,33 +1,235 @@
+)  # Static charge
+
+
+def _unit_style_factory(style: str):
+    #  NOTE: the when an angle is measured in lammps is not straightforwards. It depends not on the unit_style, but on the
+    # angle_style, dihedral_style, or improper_style. For examples, harmonic angles, k is specificed in energy/radian, but the
+    # theta_eq is written in degrees. For fourier dihedrals, d_eq is specified in degrees. When adding new styles, make sure that
+    # this behavior is accounted for when converting the specific potential_type.
+    if style == "real":
+        base_units = u.UnitSystem(
+            "lammps_real", "Å", "amu", "fs", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "kcal/mol"
+        base_units["charge"] = "elementary_charge"
+    elif style == "metal":
+        base_units = u.UnitSystem(
+            "lammps_metal", "Å", "amu", "picosecond", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "eV"
+        base_units["charge"] = "elementary_charge"
+    elif style == "si":
+        base_units = u.UnitSystem(
+            "lammps_si", "m", "kg", "s", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "joule"
+        base_units["charge"] = "coulomb"
+    elif style == "cgs":
+        base_units = u.UnitSystem(
+            "lammps_cgs", "cm", "g", "s", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "erg"
+        # Statcoulomb is strange. It is not a 1:1 correspondance to charge, with base units of
+        # mass**1/2*length**3/2*time**-1.
+        # However, assuming it is referring to a static charge and not a flux, it can be
+        # converted to coulomb units. See the registry for the unit conversion to Coulombs
+        base_units["charge"] = "Statcoulomb_charge"
+    elif style == "electron":
+        base_units = u.UnitSystem(
+            "lammps_electron", "a0", "amu", "s", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "Ehartree"
+        base_units["charge"] = "elementary_charge"
+    elif style == "micro":
+        base_units = u.UnitSystem(
+            "lammps_micro", "um", "picogram", "us", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "ug*um**2/us**2"
+        base_units["charge"] = "picocoulomb"
+    elif style == "nano":
+        base_units = u.UnitSystem(
+            "lammps_nano", "nm", "attogram", "ns", "K", "rad", registry=reg
+        )
+        base_units["energy"] = "attogram*nm**2/ns**2"
+        base_units["charge"] = "elementary_charge"
+    elif style == "lj":
+        return None
+    else:
+        raise NotYetImplementedWarning
+
+    return base_units
+
+
+# f(parameter, base_unyts, parameter_styleDict) -> converted_parameter_value
+def _parameter_converted_to_float(
+    parameter, base_unyts, conversion_factorDict=None
+):
+    """Take a given parameter, and return a float of the parameter in the given style."""
+    parameter_dims = parameter.units.dimensions * 1
```

Comment:
This was intended to create a new dimensions object that wouldn't change the parameter object in place. I believe this can get removed, as the dimensions are multiplied by the conversion factors to generate new dimensions, instead of changes being made directly to the parameter.units.dimensions.